File: Examples/Rust/src/examples/bluetooth_connection.rs
use super::connection;
use crate::helpers;
use ximu3::connection_config::*;
use ximu3::port_scanner::*;

pub fn run() {
    if helpers::yes_or_no("Search for connections?") {
        let devices = PortScanner::scan_filter(PortType::Bluetooth);

        let Some(device) = devices.first() else {
            println!("No Bluetooth connections available");
            return;
        };

        println!("Found {device}");

        let config = &device.connection_config;

        connection::run(config);
    } else {
        let config = &ConnectionConfig::BluetoothConnectionConfig(BluetoothConnectionConfig {
            port_name: "COM1".to_owned(),
        }); // replace with actual connection config

        connection::run(config);
    }
}

File: Examples/C/examples/bluetooth_connection.c
#include "../helpers.h"
#include "connection.h"
#include <stdio.h>
#include "Ximu3.h"

void bluetooth_connection() {
    if (yes_or_no("Search for connections?")) {
        const XIMU3_Devices devices = XIMU3_port_scanner_scan_filter(XIMU3_PortTypeBluetooth);

        if (devices.length == 0) {
            printf("No Bluetooth connections available\n");
            return;
        }

        printf("Found %s\n", XIMU3_device_to_string(devices.array[0]));

        const XIMU3_BluetoothConnectionConfig config = devices.array[0].bluetooth_connection_config;

        XIMU3_devices_free(devices);

        XIMU3_Connection *const connection = XIMU3_connection_new_bluetooth(config);

        run(connection);
    } else {
        const XIMU3_BluetoothConnectionConfig config = (XIMU3_BluetoothConnectionConfig){
            .port_name = "COM1",
        }; // replace with actual connection config

        XIMU3_Connection *const connection = XIMU3_connection_new_bluetooth(config);

        run(connection);
    }
}
